### PR TITLE
Feat/6412 python utils refactor

### PR DESCRIPTION
**Pull request by adurkan-amido**
#### 📲 What

Refactor Python utilities and PySparkle, organised under Datastacks.

#### 🤔 Why

So data utilities are clearly named and structured.

#### 🛠 How

Refactor and consolidate existing modules.

#### 👀 Evidence

Purpose of refactor is to restructure existing code, not change functionality / outputs.

#### 🕵️ How to test

Ensure all resources all deploy and continue to function as previously.

#### ✅ Acceptance criteria Checklist

- [x] Code peer reviewed?
- [x] Documentation has been updated to reflect the changes?
- [x] Passing all automated tests, including a successful deployment?
- [x] Passing any exploratory testing?
- [x] Rebased/merged with latest changes from development and re-tested?
- [x] Meeting the Coding Standards?

### CODE REVIEW

File: datastacks/datastacks/utils.py
Comment (adurkan-amido) [inline datastacks/datastacks/utils.py:1]: The previous content from this file is now in datastacks/datastacks/cli/utils.py

File: datastacks/tests/unit/test_utils.py
Comment (adurkan-amido) [inline datastacks/tests/unit/test_utils.py:1]: Previous version of this now in `datastacks/tests/unit/cli/test_utils.py`

File: pysparkle/README.md
Comment (adurkan-amido) [inline pysparkle/README.md:1]: Everything under `pysparkle` moved to `datastacks/datastacks/pyspark`

File: utils/config_utils.py
Comment (adurkan-amido) [inline utils/config_utils.py:1]: Moved into `datatstacks/datastacks/utils.py`

File: datastacks/datastacks-pipeline.yml
Comment (adurkan-amido) [inline datastacks/datastacks-pipeline.yml:1]: This now includes steps which were previously in `pysparkle/pysparkle-pipeline.yml`

**Review by james-horrocks — APPROVED** [review]
lgtm

**Pull request merged**